timeseries_storage/blueflood/blueflood.go

```diff
@@ -450,6 +450,44 @@ var samplerMap map[api.SampleMethod]sampler = map[api.SampleMethod]sampler{
 	},
 }
 
+// Blueflood will use the finest-grained resolution which doesn't exceed the slot limit.
+// Thus, if you request too many points, it will automatically downsample the data.
+func (b *Blueflood) AdjustTimerange(requested api.Timerange, slotLimit int) api.Timerange {
+	// In some cases, coarser-resolution data may have a shorter TTL.
+	// To accomodate these cases, it must be verified that the requested timerange will
+	// actually be present for the chosen resolution.
+	// TODO: figure out how to make this work with moving averages and timeshifts
```

Comment:
Let's work through some examples together when we get some time. Apropos of my other remark, I don't like time range adjustment as a top-level concept in the API. At least at this point. But what it sounds like it's implying is that function evaluation context is what carries this time frame in addition to some kind of deeper response object so the underlying implementation can give hints back. If that were the case, it would effectively be a pre-validation step that just happens to either return an error, or support a flag to auto-adjust ranges.

Secondly, and semi-disconnected, I'm fairly strongly opposed to data interpolation at this point unless we have a very reference implementation and a good way to signal to the user that we've synthesized data. It's possible to work with variable resolution time ranges and I'd be more apt to start thinking about how we would integrate something like that now, before we go a lot farther with this design.